docker/Dockerfile

```diff
@@ -37,11 +51,16 @@ RUN scripts/install_tools.sh
 RUN go generate && go build -tags production -o kjudge cmd/kjudge/main.go
 
 # Stage 3: Create awesome output image
-FROM ubuntu:jammy
+FROM base-ubuntu
 
-RUN apt-get update && DEBIAN_FRONTEND="noninteractive" TZ="Asia/Ho_Chi_Minh" apt-get install -y \
-    build-essential openjdk-11-jdk-headless fp-compiler python3.6 cgroup-lite python2.7 rustc golang libcap-dev # Compilers  \
-    openssl # For HTTPS support
+# Python PPA
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get install -y software-properties-common && add-apt-repository ppa:deadsnakes/ppa && \
```

Comment:
What is this PPA?

Reply:
it's a PPA for old Python distributions. It's quite reputable as far as I know.

Reply:
Installing python3.6 like the previous Dockerfile installed libcasa-python3-6, which I can't find any information on, and doesn't seem to install a "normal" python distribution (according to dpkg output).